xmppserver/src/main/java/org/jivesoftware/openfire/disco/IQDiscoInfoHandler.java

```diff
@@ -503,14 +503,15 @@ private void restoreCacheContent() {
      * @return first dataform from the collection. Null if the collection was empty.
      */
     public static DataForm getFirstDataForm(Set<DataForm> dataForms){
-        if (dataForms == null || dataForms.isEmpty() ) {
+        try {
+            if (dataForms != null && dataForms.size() > 1) {
+                Log.warn("Set Data List contains "+dataForms.size()+" DataForms."+
+                "Only the first one of the DataForms will be returned.");
+            }
+            return  dataForms.stream().filter(Objects::nonNull).findAny().orElse(null);
+        } catch (Exception e) {
```

Comment:
Why do you wrap this in a try/catch?

Ohh, line 511 can nullpointer! I didn't see that before. Maybe your previous version was better after all. :)

If you ever catch an exception, always, always, always do something with the exception - if only logging it to 'debug'. I don't think we should need a try/catch block here, by the way - just use your previous version.